model/riscv_sys_regs.sail

```diff
@@ -567,6 +567,26 @@ function retire_instruction() -> unit = {
        then minstret = minstret + 1
 }
 
+bitfield MEnvCfg : bits(64) = {
+  HADE : 61
+  /* Fill in other fields when adding support for them */
+}
+register menvcfg : MEnvCfg
+
+function legalize_menvcfg(c : MEnvCfg, high : bool, value : xlenbits) -> MEnvCfg = {
+  if not(high) & sizeof(xlen) == 32 then
+    c
+  else {
+    /* If 32bit, then high is true */
+    let full : bits(64) = match sizeof(xlen) {
+      64 => value,
+      32 => value @ 0x0000_0000
+    };
+    let c = update_HADE(c, [full[61]]);
```

Comment:
Looking at the Svadu spec I see: `When menvcfg.HADE is zero, the implementation behaves as though Svadu were not implemented. If Svadu is not implemented, menvcfg.HADE is read-only zero.` so I believe this needs to be guarded by plat_enable_svadu()

Reply:
Thanks! I needed to move `plat_enable_svadu()` out of `riscv_platform.sail` and added a note similar to the one by `get_16_random_bits` explaining it, but I made this change